src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -155,7 +157,8 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             {
                 auto index = winrt::unbox_value<uint8_t>(tag);
                 CurrentColorScheme().SetColorTableEntry(index, args.NewColor());
-                _CurrentColorTable.GetAt(index).Color(args.NewColor());
+                _CurrentNonBrightColorTable.GetAt(index).Color(args.NewColor());
+                _CurrentBrightColorTable.GetAt(index).Color(args.NewColor());
             }
```

Comment:
(read this comment last. I suggest you figure out the other two comments first.)

So, each `ColorTableEntry` has an index property. This index is the index in the color table. So if you look at `TableColorNames`, this should help you out. A color table entry with the index of 0 is our representation of "black", 1st is "red", etc...

So here we have to do that `8` trick backwards. The 0th color in `_CurrentNonBrightColorTable` is black, which is the 0th color of the color scheme's color table. 1st non-bright is red --> 1st in color scheme color table. Whereas the 0th in the bright color table is actually the 8th in the color scheme's color table (bright black needs to map to bright black). etc.

Reply:
Okay, I've got the others down, but I'm not quite sure I understand this one. Would you please be more specific? Thanks in advance

Reply:
@carlos-zamora Here's my current code, however, it doesn't seem to work:

```cpp
void ColorSchemes::ColorPickerChanged(IInspectable const& sender,
                                          ColorChangedEventArgs const& args)
    {
        if (auto picker = sender.try_as<ColorPicker>())
        {
            if (auto tag = picker.Tag())
            {
                auto index = winrt::unbox_value<uint8_t>(tag);
                if (index < 8)
                {
                    CurrentColorScheme().SetColorTableEntry(index, args.NewColor());
                }
                else if (index > 8)
                {
                    CurrentColorScheme().SetColorTableEntry(index + 8, args.NewColor());
                }
                _CurrentNonBrightColorTable.GetAt(index).Color(args.NewColor());
                _CurrentBrightColorTable.GetAt(index).Color(args.NewColor());
            }
        }
    }
```

Reply:
So, we shouldn't really be changing our interaction with `CurrentColorScheme()` in this PR. `CurrentColorScheme` is the color scheme object from the settings model. This function gets called when the user picks a color from the flyout (i.e. changing "black" to #FFFFFF). So, here, our interaction with the `CurrentColorScheme` is that we're updating the settings model.

`_CurrentColorTable`, on the other hand, is solely a part of the Settings UI. Here, we're updating which color to show in the UI. Before, it was all one color table. But now, we've split the table into 2 (brights and non-brights). So the if-statements should determine which of those color tables to modify. Whereas the settings model can stay the same as it was before because the index is the same.

Reply:
@carlos-zamora Thank you, do you mean something like this (The two different colour pickers will point to the appropriate method)?
```cpp
    void ColorSchemes::NonBrightColorPickerChanged(IInspectable const& sender,
                                          ColorChangedEventArgs const& args)
    {
        if (auto picker = sender.try_as<ColorPicker>())
        {
            if (auto tag = picker.Tag())
            {
                auto index = winrt::unbox_value<uint8_t>(tag);
                CurrentColorScheme().SetColorTableEntry(index, args.NewColor());
                _CurrentNonBrightColorTable.GetAt(index).Color(args.NewColor());
            }
        }
    }

    void ColorSchemes::BrightColorPickerChanged(IInspectable const& sender,
                                                   ColorChangedEventArgs const& args)
    {
        if (auto picker = sender.try_as<ColorPicker>())
        {
            if (auto tag = picker.Tag())
            {
                auto index = winrt::unbox_value<uint8_t>(tag);
                CurrentColorScheme().SetColorTableEntry(index, args.NewColor());
                _CurrentBrightColorTable.GetAt(index).Color(args.NewColor());
            }
        }
    }
```

Reply:
Sure, something like that is close.

There's a small bug in the `BrightColorPickerChanged` function. Remember, all of the bright colors have an index >= 8. So bright black is 8, bright red is 9, etc. Since the tag for bright colors will always be >= 8, you won't be interacting with any entries < 8 in `_CurrentBrightColorTable`. Instead, offset it so that bright black is on index 0 for `_CurrentBrightColorTable`, bright red is on 1, etc...

Some meta advice: please try implementing your solutions and debugging through them yourself. I'm happy to help when you get stuck, but I've explained how the color table works a few times in this thread alone now.

Reply:
@carlos-zamora Sorry if I was a bit annoying with all the questions. I do understand how the colour table works, but I'm stuck on how to offset it. Thank you very much for all your help.